rules/windows/discovery_post_exploitation_external_ip_lookup.toml

```diff
@@ -62,74 +62,53 @@ references = [
 risk_score = 21
 rule_id = "1d72d014-e2ab-4707-b056-9b96abe7b511"
 severity = "low"
-tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Discovery", "Resources: Investigation Guide", "Data Source: Elastic Defend"]
+tags = ["Domain: Endpoint",
+        "OS: Windows",
+        "Use Case: Threat Detection",
+        "Tactic: Discovery",
+        "Resources: Investigation Guide",
+        "Data Source: Elastic Defend"
+        ]
 timestamp_override = "event.ingested"
-type = "eql"
+type = "new_terms"
 
 query = '''
-network where host.os.type == "windows" and network.protocol == "dns" and
-    process.name != null and user.id not in ("S-1-5-19", "S-1-5-20") and
-    event.action == "lookup_requested" and
-    /* Add new external IP lookup services here */
-    dns.question.name :
-    (
-        "*api.ipify.org",
-        "*freegeoip.app",
-        "*checkip.amazonaws.com",
-        "*checkip.dyndns.org",
-        "*freegeoip.app",
-        "*icanhazip.com",
-        "*ifconfig.*",
-        "*ipecho.net",
-        "*ipgeoapi.com",
-        "*ipinfo.io",
-        "*ip.anysrc.net",
-        "*myexternalip.com",
-        "*myipaddress.com",
-        "*showipaddress.com",
-        "*whatismyipaddress.com",
-        "*wtfismyip.com",
-        "*ipapi.co",
-        "*ip-lookup.net",
-        "*ipstack.com"
-    ) and
-    /* Insert noisy false positives here */
-    not process.executable :
-    (
-      "?:\\Program Files\\*.exe",
-      "?:\\Program Files (x86)\\*.exe",
-      "?:\\Windows\\System32\\WWAHost.exe",
-      "?:\\Windows\\System32\\smartscreen.exe",
-      "?:\\Windows\\System32\\MicrosoftEdgeCP.exe",
-      "?:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\*\\MsMpEng.exe",
-      "?:\\Users\\*\\AppData\\Local\\Google\\Chrome\\Application\\chrome.exe",
-      "?:\\Users\\*\\AppData\\Local\\Programs\\Fiddler\\Fiddler.exe",
-      "?:\\Users\\*\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe",
-      "?:\\Users\\*\\AppData\\Local\\Microsoft\\OneDrive\\OneDrive.exe"
-    )
+host.os.type:windows and event.category:network and network.protocol:dns and process.name:* and 
+event.action:lookup_requested and dns.question.name:(
```

Comment:
I wonder if `dns.question.name` is normalized to lowercase. If not, I think this change is still worth the risk of FN

Reply:
oooh .... or

Leave the original rule, but strip it down and convert it to a BBR

then for this new terms rule, look in the alerts index and make the query search for this alert


so:

## this rule

```sql
network where host.os.type == "windows" and network.protocol == "dns" and
    process.name != null and user.id not in ("S-1-5-19", "S-1-5-20") and
    event.action == "lookup_requested" and
    /* Add new external IP lookup services here */
    dns.question.name :
    (
        "*api.ipify.org",
        "*freegeoip.app",
        "*checkip.amazonaws.com",
        "*checkip.dyndns.org",
        "*freegeoip.app",
        "*icanhazip.com",
        "*ifconfig.*",
        "*ipecho.net",
        "*ipgeoapi.com",
        "*ipinfo.io",
        "*ip.anysrc.net",
        "*myexternalip.com",
        "*myipaddress.com",
        "*showipaddress.com",
        "*whatismyipaddress.com",
        "*wtfismyip.com",
        "*ipapi.co",
        "*ip-lookup.net",
        "*ipstack.com"
```

## new terms rule

```sql
event.kind signal and kibana.alert.rule.rule_id:"1d72d014-e2ab-4707-b056-9b96abe7b511" and
  not process.executable.caseless:
    (
      C:\\Program Files\\*.exe or
      C:\\Program Files (x86)\\*.exe or
      C:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\*\\MsMpEng.exe
    )
```

new terms

```
 ["agent.id", "user.name", "process.executable"]
```

Reply:
you could potentially consider this approach for a few of the conversions as well